docs/proposals/ose-rebase.md

```diff
@@ -0,0 +1,53 @@
+## Rebase OSE on Origin
+
+Today, code in OSE lands by fetching and merging Origin manually. OSE-specific changes
+are scattered in the repo and it's hard to reason about any conflicts. This doc proposes
+moving all the changes in OSE that diverge from Origin into a set of commits that will sit
+on top of the Origin code[0] and a Jenkins job that will be rebasing master on a daily basis.
+Rebases are going to continue to happen only for the master branch.
+
+There are two types of commits that need to be carried on top of Origin:
+1. long-term carries related to branding and release tooling (master and release branches)
+2. CVEs and other backported bugfixes (release branches only)
+
+The Origin repository will be set as a remote `upstream` branch in OSE and in ose/master we can
+simply rebase upstream/master. If any conflicts are hit, manual resolution will be required. In
+the future, we can extend the conflict resolution mechanism in Jenkins to send an e-mail either
+to the author and committer of the carry commit or we can store e-mails in the commit description
+and use those instead.
+
+A nightly build of OSE should follow the steps below in order to rebase on top of Origin:
+```
+git clone git@github.com/openshift/ose.git
+pushd ose
+git remote add upstream git@github.com/openshift/origin.git
+git fetch upstream
+git checkout master
+git rebase upstream/master
+```
+
+New releases should be cut as they already are today (a mixture of git and tito). CVEs and bugfixes
+should continue to land in release branches as they already are today too. CVEs and important bugfixes
+will continue to be backported back to Origin manually but we may eventually build a script that will
+create git commits locally automatically, eg. hack/cherry-pick-from-ose.sh openshift/ose/pull/578.
+
+Common conflicts can rise when new changes step on branding or tooling code. Manual resolution is the
+only thing we can do in such cases but we can reduce the amount of conflicts by using build flags that
+will replace branding instances depending on the repository that is being built. We should also investigate
+whether we can stop checking in various code like tito diffs, generated docs[1], and the ose-images script.
+Avoiding those diffs will reduce even further the number of conflicts.
+
+Naming conventions we can use in commit messages in OSE:
+* commits that need to be carried long-term such as branding or release tooling changes will start with [CARRY]
+* commits that are fixing branding or release tooling changes will start with [SQUASH]. The rebase process
+should be responsible for squashing those commits appropriately into the [CARRY] commits.
+* bugfixes and CVEs can contain a [FIX] tag. We would have to enforce these when developers open PRs against the
```

Comment:
I see some cases where we might do a rebase but agree that it's going to be rare and manual so we don't need this